### PR TITLE
com.google.santa.daemon: async es message handling

### PR DESCRIPTION
**Pull request by tburgin**
* Have the EventProvider create and dispatch messages on appropriate queues
* Refactor how copying es message -> santa message fields are done, hopefully in a safer way.
* Fix Xcode project

### CODE REVIEW

File: Source/santad/EventProviders/SNTEndpointSecurityManager.mm [hunk 1]
```diff
@@ -127,94 +149,103 @@ - (void)messageHandler:(const es_message_t *)m API_AVAILABLE(macos(10.15)) {
...
-      sm.path[l] = '\0';
-
       break;
     }
     case ES_EVENT_TYPE_NOTIFY_CLOSE: {
-      if (!m->event.close.modified) return;
+      if (!m->event.close.modified) {
+        es_free_message(m);
+        return;
+      }
```
Comment (russellhancox) [inline Source/santad/EventProviders/SNTEndpointSecurityManager.mm]: This path will never be hit, is it just a last ditch check?

File: Source/santad/EventProviders/SNTEndpointSecurityManager.mm [hunk 2]
```diff
@@ -127,94 +149,103 @@ - (void)messageHandler:(const es_message_t *)m API_AVAILABLE(macos(10.15)) {
...
+  if ([self populateBufferFromESFile:targetFile buffer:sm.path size:sizeof(sm.path)]
+      && m->event_type == ES_EVENT_TYPE_AUTH_EXEC) {
+    LOGE(@"path is truncated, deny: %s", sm.path);
+    es_respond_auth_result(self.client, m, ES_AUTH_RESULT_DENY, false);
+    es_free_message(m);
+    return;
+  }
+
+  // Filter file op events matching the prefix tree.
   if (!(m->event_type == ES_EVENT_TYPE_AUTH_EXEC || m->event_type == ES_EVENT_TYPE_NOTIFY_EXEC) &&
```
Comment (russellhancox) [inline Source/santad/EventProviders/SNTEndpointSecurityManager.mm]: How come the prefix tree is applying to auth events?

File: Source/santad/EventProviders/SNTEndpointSecurityManager.mm (hunk 1 above)
Comment (tburgin) [reply Source/santad/EventProviders/SNTEndpointSecurityManager.mm]: Removed

File: Source/santad/EventProviders/SNTEndpointSecurityManager.mm (hunk 2 above)
Comment (tburgin) [reply Source/santad/EventProviders/SNTEndpointSecurityManager.mm]: It doesn't apply to any exec event, note the negation of the (||) clause.

Comment (russellhancox) [reply Source/santad/EventProviders/SNTEndpointSecurityManager.mm]: Ah I missed that

**Pull request merged**